packages/shared/lib/services/environment.service.ts

```diff
@@ -156,51 +96,56 @@ class EnvironmentService {
         return uuid;
     }
 
-    async getAccountIdAndEnvironmentIdByPublicKey(publicKey: string): Promise<{ accountId: number; environmentId: number } | null> {
+    async getAccountAndEnvironmentByPublicKey(publicKey: string): Promise<{ account: Account; environment: Environment } | null> {
         if (!isCloud) {
-            const environmentVariables = Object.keys(process.env).filter((key) => key.startsWith('NANGO_PUBLIC_KEY_')) || [];
+            const environmentVariables = Object.keys(process.env).filter((key) => key.startsWith('NANGO_PUBLIC_KEY_'));
             if (environmentVariables.length > 0) {
                 for (const environmentVariable of environmentVariables) {
                     const envPublicKey = process.env[environmentVariable] as string;
 
-                    if (envPublicKey === publicKey) {
-                        const env = environmentVariable.replace('NANGO_PUBLIC_KEY_', '').toLowerCase();
-                        // This key is set dynamically and does not exists in database
-                        const environment = await db.knex.select('*').from<Environment>(TABLE).where({ name: env }).first();
-
-                        if (!environment) {
-                            return null;
-                        }
-
-                        return { accountId: environment.account_id, environmentId: environment.id };
+                    if (envPublicKey !== publicKey) {
+                        continue;
                     }
+                    const envName = environmentVariable.replace('NANGO_PUBLIC_KEY_', '').toLowerCase();
+                    // This key is set dynamically and does not exists in database
+                    const env = await db.knex.select<Pick<Environment, 'account_id'>>('account_id').from<Environment>(TABLE).where({ name: envName }).first();
+                    if (!env) {
+                        return null;
+                    }
+
+                    return this.getAccountAndEnvironment({ accountId: env.account_id, envName });
                 }
             }
         }
```

Comment:
is this function logic the same  as  `getAccountAndEnvironmentBySecretKey` but with a different key?

Reply:
yes, I initially planned on refactoring this but reverted while trying to keep the unrelated change to a minimum. I'm sure there is a method that could do both